demos/interactive_face_detection_demo/cpp/main.cpp

```diff
@@ -2,228 +2,311 @@
+DEFINE_string(i, "", input_msg);
 
-    if (FLAGS_m.empty()) {
-        throw std::logic_error("Parameter -m is not set");
-    }
-    return true;
-}
+constexpr char loop_msg[] = "Enable reading the input in a loop";
+DEFINE_bool(loop, false, loop_msg);
 
-int main(int argc, char *argv[]) {
-    try {
-        PerformanceMetrics metrics;
+constexpr char output_msg[] = "Name of the output file(s) to save";
+DEFINE_string(o, "", output_msg);
 
-        // ------------------------------ Parsing and validating of input arguments --------------------------
-        if (!ParseAndCheckCommandLine(argc, argv)) {
-            return 0;
-        }
+constexpr char limit_msg[] = "Number of frames to store in output. If 0 is set, all frames are stored. Default is 1000";
+DEFINE_uint32(limit, 1000, limit_msg);
+
+constexpr char help_msg[] = "show this help message and exit";
+DEFINE_bool(h, false, help_msg);
+
+constexpr char face_detection_model_msg[] = "Path to an .xml file with a trained Face Detection model";
+DEFINE_string(m, "", face_detection_model_msg);
+
+constexpr char age_gender_model_msg[] = "Path to an .xml file with a trained Age/Gender Recognition model";
+DEFINE_string(m_ag, "", age_gender_model_msg);
+
+constexpr char head_pose_model_msg[] = "Path to an .xml file with a trained Head Pose Estimation model";
+DEFINE_string(m_hp, "", head_pose_model_msg);
+
+constexpr char emotions_model_msg[] = "Path to an .xml file with a trained Emotions Recognition model";
+DEFINE_string(m_em, "", emotions_model_msg);
+
+constexpr char facial_landmarks_model_msg[] = "Path to an .xml file with a trained Facial Landmarks Estimation model";
+DEFINE_string(m_lm, "", facial_landmarks_model_msg);
+
+constexpr char antispoofing_model_msg[] = "Path to an .xml file with a trained Antispoofing Classification model";
+DEFINE_string(m_am, "", antispoofing_model_msg);
+
+constexpr char device_msg[] =
+    "Specify a target device to infer on (the list of available devices is shown below). "
+    "Use \"-d HETERO:<comma-separated_devices_list>\" format to specify "
+    "HETERO plugin. "
+    "Use \"-d MULTI:<comma-separated_devices_list>\" format to specify MULTI plugin. "
+    "The application looks for a suitable plugin for the specified device."
+    "Default is CPU";
+DEFINE_string(d, "CPU", device_msg);
 
-        // --------------------------- 1. Loading Inference Engine -----------------------------
+constexpr char thresh_output_msg[] = "Probability threshold for detections. Default is 0.5";
+DEFINE_double(t, 0.5, thresh_output_msg);
+
+constexpr char bb_enlarge_coef_output_msg[] = "Coefficient to enlarge/reduce the size of the bounding box around the detected face. Default is 1.2";
+DEFINE_double(bb_enlarge_coef, 1.2, bb_enlarge_coef_output_msg);
+
+constexpr char raw_output_msg[] = "Output inference results as raw values";
+DEFINE_bool(r, false, raw_output_msg);
+
+constexpr char no_show_msg[] = "Don't show output";
+DEFINE_bool(no_show, false, no_show_msg);
```

Comment:
I think we can replace `--no_show` with just `-n` now to keep it one char, while keeping `--noshow` alternative for Python. @vladimir-dudnik, what do you think?

Reply:
@Wovchena are you saying we will not have long name options in C++ anymore?

Reply:
For numbers we explicitly define `nsomething` flag

Reply:
ok, but what is the rationale of this change, why you want to do this?

Reply:
one symbol is shorter :)

Reply:
and less intuitive

Reply:
ok. @ieliz, please apply the same trick with autogenerated no prefix